### ISSUE TITLE
dependency :extension ignores archives that are not jar or zip

### ISSUE THREAD

**Issue by blueberry** [technomancy/leiningen#1789]
Hi,

I am trying to use nar archives, which are basically jars that contain native libraries created by nar-maven-plugin.
I discovered that if I only rename libraryX.nar to libraryX.jar or libraryX.zip in .m2 repository, the library is included in the classpath of my lein project with no further intervention. libraryX.zip needs :extension "zip" in :dependencies section.

It logically follows that it should also work with libraryX.nar and :extension "nar" but it does not - the dependency is accepted but not added to the classpath. It seems that lein filters everything except zip or jar when it reaches a regexp in the resolve-dependencies function in classpath.clj.

Is there a way in leiningen 2.5.0 to bypass this through configuration or by writing a plugin?
If not, is it possible that the restriction be relaxed in the next leiningen release?

**Comment by flosell** [comment]
+1

**Comment by hyPiRion** [comment]
Sorry for the delayed response here: Is it a standard convention to end these libraries with `.nar`?

**Comment by blueberry** [comment]
yes, it is a standard in a sense that the nar-maven-plugin does this by default and some of the functionality of the resulting archive depend on the .nar extension. Also, there is no way to change the extension in configuration.

**Comment by flosell** [comment]
For me it's not `.nar` but `.tar`: 

My application needs to use some files that are published to our internal nexus-repository as a tar-file. Would be neat if I could just declare a dependency to this artifact and have them on the classpath.
